### PR TITLE
JenkinsFile Fix

### PR DESCRIPTION
**Pull request by jayahariv**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -7,20 +7,41 @@ pipeline {
...
-                    git clone https://github.com/couchbaselabs/${env.PRODUCT}.git
-                    pushd ${env.PRODUCT}
+                    #!/bin/bash
+                    set -e
+                    shopt -s extglob dotglob
+
+		    # move PR related repo to tmp folder
+		    mkdir tmp
+                    mv !(tmp) tmp
+                    git clone https://github.com/couchbaselabs/${env.PRODUCT}.git --branch $CHANGE_TARGET
```
Comment (pasin) [inline Jenkinsfile]: I think moving this line under # restructure folders to be together with `mv couchbase-lite-ios-ee/* ` feel less confusing.

Comment (pasin) [reply Jenkinsfile]: env.PRODUCT is only used once here so I don't think we need to declare it as a PRODUCT environment.

File: Jenkinsfile [hunk 2]
```diff
@@ -7,20 +7,41 @@ pipeline {
...
+
+		    # restructure folders
+		    mv couchbase-lite-ios-ee/* .
+		    mv tmp/* couchbase-lite-ios
+
+		    # update the lite-core-EE
+		    pushd couchbase-lite-core-EE
+		    git submodule update --init ./
```
Comment (pasin) [inline Jenkinsfile]: couchbase-lite-core-EE doesn't have any submodules so we can remove this section entirely. If couchbase-lite-core-EE starts to have submodule, we can add this.

Comment (jayahariv) [reply Jenkinsfile]: if not updated the submodule, then we will get this error
http://mobile.jenkins.couchbase.com/blue/organizations/jenkins/_cbl_github_pipeline%2Fcouchbase-lite-ios/detail/PR-2561/8/pipeline

Comment (pasin) [reply Jenkinsfile]: It really doesn't have submodules so `git submodule update --init ./` will do nothing. Please double check.

File: Jenkinsfile [hunk 3]
```diff
@@ -1,26 +1,39 @@
...
+		    # move PR related repo to tmp folder
+		    mkdir tmp
+                    mv !(tmp) tmp
+                    git clone https://github.com/couchbaselabs/couchbase-lite-ios-ee.git --branch $CHANGE_TARGET
+
+		    # restructure folders
+		    mv couchbase-lite-ios-ee/* .
+		    mv tmp/* couchbase-lite-ios
+		    
+		    # submodule update inside lite-ios
```
Comment (bmeike) [inline Jenkinsfile]: I don't really know what's going on, here, but it seems odd that you check out the root directory without submodules and then go into one of the submodules and init it recursively.  Maybe a little comment that says what's going on, or something?

Comment (pasin) [reply Jenkinsfile]: I think @bmeike just pointed out the issue. After cloning the couchbase-lite-ios-ee, the submodule needs to be init and update. The following can be done when cloning couchbase-lite-ios-ee.

```git clone https://github.com/couchbaselabs/couchbase-lite-ios-ee.git --branch $CHANGE_TARGET -- recursive```

**Comment by jayahariv** [comment]
please check again?

File: Jenkinsfile [hunk 4]
```diff
@@ -1,26 +1,45 @@
...
-                    ./Scripts/prepare_project.sh
-                    cd couchbase-lite-ios
-		    git checkout master
-		    git pull origin master
-                    popd
+		    popd
+
+		    # restructure folders
+		    mv -v couchbase-lite-ios-ee/* .
+		    rsync -a tmp/ couchbase-lite-ios/
```
Comment (pasin) [inline Jenkinsfile]: I'm not sure if --delete is needed to completely replace all content inside couchbase-lite-ios. 

Alternatively but two commands,
```
rm -rf couchbase-lite-ios && mv tmp couchbase-lite-ios
```

File: Jenkinsfile [hunk 5]
```diff
@@ -1,26 +1,45 @@
...
                     git submodule update --init --recursive
-                    ./Scripts/prepare_project.sh
-                    cd couchbase-lite-ios
-		    git checkout master
-		    git pull origin master
-                    popd
+		    popd
+
+		    # restructure folders
+		    mv -v couchbase-lite-ios-ee/* .
```
Comment (pasin) [inline Jenkinsfile]: Do you intend to do verbose mode or is it for debugging?

Comment (jayahariv) [reply Jenkinsfile]: not intended, it was for debugging

File: Jenkinsfile (hunk 4 above)
Comment (jayahariv) [reply Jenkinsfile]: i tried `-f ` flag, but not worked. it should force replace all contents. I will try with the two commands, which will be easy to understand.

File: Jenkinsfile (hunk 5 above)
Comment (jayahariv) [inline Jenkinsfile]:
```suggestion
		    mv couchbase-lite-ios-ee/* .
```

File: Jenkinsfile (hunk 4 above)
Comment (jayahariv) [inline Jenkinsfile]:
```suggestion
		    rm -rf couchbase-lite-ios && mv tmp couchbase-lite-ios
```

File: Jenkinsfile [hunk 6]
```diff
@@ -1,26 +1,45 @@
...
+		    mv couchbase-lite-ios-ee/* .
+		    rm -rf couchbase-lite-ios && mv tmp couchbase-lite-ios
+		    
+		    # submodule update inside lite-ios
+		    pushd couchbase-lite-ios
+                    git submodule update --init --recursive
+		    popd
+
+		    # remove tmp folders
+		    rm -rf tmp/*
```
Comment (pasin) [inline Jenkinsfile]: tmp has already been renamed to couchbase-lite-ios in line 24.  The rest is good.

Comment (jayahariv) [inline Jenkinsfile]:
```suggestion
		    rmdir tmp
```

File: Jenkinsfile [hunk 7]
```diff
@@ -1,26 +1,45 @@
...
+		    mv couchbase-lite-ios-ee/* .
+		    rm -rf couchbase-lite-ios && mv tmp couchbase-lite-ios
+		    
+		    # submodule update inside lite-ios
+		    pushd couchbase-lite-ios
+                    git submodule update --init --recursive
+		    popd
+
+		    # remove tmp folders
+		    rmdir tmp
```
Comment (jayahariv) [inline Jenkinsfile]:
```suggestion
```

**Pull request merged**